sqlmesh/lsp/main.py

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+"""A Language Server Protocol (LSP) server for SQL with SQLMesh integration."""
+
+import logging
+import typing as t
+from contextlib import suppress
+from pathlib import Path
+
+from sqlmesh.core.context import Context
+from lsprotocol import types
+from pygls.server import LanguageServer
+from pygls.workspace import TextDocument
+from sqlmesh._version import __version__
+from sqlmesh.core.model import Model
+
+logger = logging.getLogger(__name__)
+
+CONTEXTS: t.Dict[str, Context] = {}
+"""A mapping of workspace paths to SQLMesh contexts."""
+
+PATHS_TO_MODELS: t.Dict[str, t.Tuple[Context, Model]] = {}
+"""A mapping of file paths to SQLMesh (context, model) tuples."""
+
+server = LanguageServer("sqlmesh_lsp", __version__)
+
+_CACHE: t.Set[str] = set()
+"""A cache of URIs for which we have already ensured a context exists."""
+
+
+def ensure_context_for_document(document: TextDocument) -> TextDocument:
```

Comment:
I suggest adding some line breaks here and in `formatting`, the continuous block of text is hard to read.

Reply:
This should be now somewhat improved.